File: inventory-service/src/main/java/com/example/inventory_service/service/InventoryService.java
package com.example.inventory_service.service;

import com.ecommerce.common.dto.InventoryResponse;
import com.example.inventory_service.dto.CreateInventoryRequest;
import com.ecommerce.common.dto.InventoryQuantityRequest;

public interface InventoryService {

    InventoryResponse createInventory(CreateInventoryRequest request);

    InventoryResponse getInventory(Long productId);

    InventoryResponse increaseStock(Long productId, InventoryQuantityRequest request);

    InventoryResponse decreaseStock(Long productId, InventoryQuantityRequest request);

    InventoryResponse reserveStock(Long productId, InventoryQuantityRequest request);

    InventoryResponse releaseStock(Long productId, InventoryQuantityRequest request);

    InventoryResponse confirmReservation(Long productId, InventoryQuantityRequest request);

    void createInventoryForProduct(Long productId);
}

File: ecommerce-frontend/src/services/inventoryService.js
import { apiGet, apiPost } from "./api";

function createInventory(request) {
  return apiPost("/inventory", request);
}

function getInventory(productId) {
  return apiGet(`/inventory/${productId}`);
}

function increaseStock(productId, quantity) {
  return apiPost(
    `/inventory/${productId}/increase`,
    {
      quantity
    }
  );
}

function decreaseStock(productId, quantity) {
  return apiPost(
    `/inventory/${productId}/decrease`,
    {
      quantity
    }
  );
}

function reserveStock(
  productId,
  quantity,
  reservationId = null
) {
  return apiPost(
    `/inventory/${productId}/reserve`,
    {
      quantity,
      reservationId
    }
  );
}

function releaseStock(
  productId,
  quantity,
  reservationId = null
) {
  return apiPost(
    `/inventory/${productId}/release`,
    {
      quantity,
      reservationId
    }
  );
}

function confirmReservation(
  productId,
  quantity,
  reservationId = null
) {
  return apiPost(
    `/inventory/${productId}/confirm`,
    {
      quantity,
      reservationId
    }
  );
}

export {
  createInventory,
  getInventory,
  increaseStock,
  decreaseStock,
  reserveStock,
  releaseStock,
  confirmReservation
};